src/envoy/auth/jwt.h

```diff
@@ -0,0 +1,48 @@
+#ifndef PROXY_JWT_H
+#define PROXY_JWT_H
+
+#include "openssl/evp.h"
+#include "rapidjson/document.h"
+
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace Envoy {
+namespace Http {
+namespace Auth {
+
+namespace Base64url {
+
+std::string decode(std::string input);
+
+}  // Base64url
+
+namespace Util {
+
+uint8_t* unsigned_c_str(const std::string& str);
+std::vector<std::string> split(std::string str, char c);
+
+}  // Util
+
+class Jwt {
+ public:
+  std::pair<bool, rapidjson::Document*> decode(const std::string& jwt,
```

Comment:
If you the 1st of the pair is `false` iff the document is `nullptr` then you don't need the `bool`.